test/e2e/delete_test.go

```diff
@@ -250,6 +244,52 @@ func (tc *testContext) testWindowsNodeDeletion(t *testing.T) {
 	require.NoError(t, err, "could not delete test namespace")
 }
 
+// deployWorkloadsToDrain creates workloads that require special options to drain
+func (tc *testContext) deployWorkloadsToDrain() error {
+	// Deploy a DaemonSet and wait until its pods have been made ready on each Windows node. DaemonSet pods cannot be
+	// drained from a Node.
+	// Doing this ensures that WMCO is able to handle containers being present on the instance when deconfiguring.
+	ds, err := tc.deployNOOPDaemonSet()
+	if err != nil {
+		fmt.Errorf("error creating DaemonSet")
+	}
+	defer tc.client.K8s.AppsV1().DaemonSets(ds.GetNamespace()).Delete(context.TODO(), ds.GetName(),
+		meta.DeleteOptions{})
+	err = tc.waitUntilDaemonsetScaled(ds.GetName(), int(gc.numberOfMachineNodes+gc.numberOfBYOHNodes))
+	if err != nil {
+		fmt.Errorf("error waiting for DaemonSet pods to become ready")
+	}
+
+	volumes := []core.Volume{{
+		Name: "test-volume",
+		VolumeSource: core.VolumeSource{EmptyDir: &core.EmptyDirVolumeSource{
+			Medium: core.StorageMediumDefault}}},
+	}
+	volumeMounts := []core.VolumeMount{{Name: "test-volume", MountPath: "/test/"}}
+	var affinity *core.Affinity
+	for _, node := range gc.allNodes() {
+		affinity, err = getAffinityForNode(&node)
```

Comment:
affinity is being overwritten in this loop and will only take the final value.
I don't think passing in an affinity is necessary. You should be able to pass in nil.
please check `createWindowsServerDeployment` for the expected behavior when nil is passed in.